src/boot/efi/boot.c

```diff
@@ -1954,34 +1955,8 @@ static void config_entry_add_windows(Config *config, EFI_HANDLE *device, EFI_FIL
 
         /* Try to find a better title. */
         err = file_read(root_dir, L"\\EFI\\Microsoft\\Boot\\BCD", 0, 100*1024, &bcd, &len);
-        if (!EFI_ERROR(err)) {
-                static const CHAR16 *versions[] = {
-                        L"Windows 11",
-                        L"Windows 10",
-                        L"Windows 8.1",
-                        L"Windows 8",
-                        L"Windows 7",
-                        L"Windows Vista",
-                };
-
-                CHAR8 *p = bcd;
-                while (!title) {
-                        CHAR8 *q = mempmem_safe(p, len, versions[0], STRLEN(L"Windows "));
-                        if (!q)
-                                break;
-
-                        len -= q - p;
-                        p = q;
-
-                        /* We found the prefix, now try all the version strings. */
-                        for (UINTN i = 0; i < ELEMENTSOF(versions); i++) {
-                                if (memory_startswith(p, len, versions[i] + STRLEN("Windows "))) {
-                                        title = versions[i];
-                                        break;
-                                }
-                        }
-                }
-        }
+        if (!EFI_ERROR(err))
+                title = get_bcd_title((UINT8 *) bcd, len);
```

Comment:
Why this cast is necessary?

Reply:
Fedora again:
`error: pointer targets in passing argument 1 of "get_bcd_title" differ in signedness`

Reply:
Ah, `UINT8*` vs `CHAR8*`. Thanks.